contracts/Staking.sol

```diff
@@ -1,4 +1,5 @@
 pragma solidity 0.4.24;
+pragma experimental ABIEncoderV2;
```

Comment:
My only concern with this is that it is experimental in fact. 

For instance, there was a serious [bug disclosure](https://blog.ethereum.org/2019/03/26/solidity-optimizer-and-abiencoderv2-bug/) some months ago and AFAIK it was fixed for `0.5.x` but they didn't mention anything about `0.4.x` versions. However, they [released](https://github.com/ethereum/solidity/releases/tag/v0.4.26) some `ABIEncoderV2` bug-fixes in `0.4.26` 3 days ago, I think you get the idea.